test/test_sync.cpp

```diff
@@ -6140,6 +6140,8 @@ TEST_IF(Sync_CollectionInMixed, sync::SYNC_SUPPORTS_NESTED_COLLECTIONS)
         dict->insert_collection("list", CollectionType::List);
         auto l = dict->get_list("list");
         l->add(5);
+        l->insert_collection(1, CollectionType::List);
+        l->get_list(1)->add(7);
```

Comment:
can we add some checks that the nested list (and dictionary) returns the correct table/colkey/etc?

Reply:
Done